src/app/modules/item/pages/item-progress/item-progress.component.ts

```diff
@@ -30,24 +28,18 @@ export class ItemProgressComponent implements OnChanges {
       return;
     }
 
-    this.type$.next(this.itemData.item.type);
+    this.recomputeSelector(this.itemData.item.type);
   }
 
-  private getSelectors$(): Observable<'none' | 'withUserProgress' | 'withGroupProgress'> {
-    return this.type$.pipe(
-      filter(type => !!type),
-      switchMap(type => this.sessionService.watchedGroup$.pipe(
-        map(watchedGroup => {
-          if (!watchedGroup && [ 'Task', 'Course' ].includes(type)) {
-            return 'none';
-          } else if (!watchedGroup && type === 'Chapter') {
-            return 'withUserProgress';
-          } else {
-            return 'withGroupProgress';
-          }
-        })
-      )),
-      delay(0)
-    );
+  private recomputeSelector(type: ItemType): void {
+    const isCurrentWatching = this.sessionService.isCurrentWatching;
+
+    if (!isCurrentWatching && [ 'Task', 'Course' ].includes(type)) {
+      this.selectors = 'none';
+    } else if (!isCurrentWatching && type === 'Chapter') {
```

Comment:
I prefer you use `![ 'Task', 'Course' ].includes(type) ` here, to have something more "complete". And skills if used, would be closer to chapter anyway.

Reply:
Fixed